akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolGateway.scala

```diff
@@ -25,6 +25,8 @@ import scala.concurrent.{ Future, Promise }
  */
 private[http] final class PoolGateway(gatewayRef: ActorRef, val hcps: HostConnectionPoolSetup, val gatewayId: GatewayIdentifier)(implicit fm: Materializer) {
 
+  private var shutdownFutureOpt: Option[Future[Done]] = None
```

Comment:
Perhaps we should create the `Promise` here, as a `val`?

Reply:
Thanks for review! I guess I will add a new spec for this class and will cover this improvement with tests.